api/bucket.go

```diff
@@ -225,7 +225,19 @@ func CreateBucket(c *gin.Context) {
 		createBucketRequest.SecretId = secret.GenerateSecretIDFromName(createBucketRequest.SecretName)
 	}
 
-	cloudType, err := determineCloudProviderFromRequest(createBucketRequest)
+	retrievedSecret, err := secret.Store.Get(organization.ID, createBucketRequest.SecretId)
+	if err != nil {
+		if strings.Contains(err.Error(), "there's no secret with this id") {
```

Comment:
Is there no existing utility method in any of the secret-related packages that does this check? Then there should be!

Reply:
I think here you can check for `secret.ErrSecretNotExists` .

Reply:
👍